### DISCUSSION TITLE
Paragraphs are missing in transcript response.

### DISCUSSION THREAD

**Discussion by liechtym** [deepgram/community#273] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

How to reproduce:

```python
from deepgram import Deepgram
from io import BytesIO

dg_key = "[REMOVED]"
dg = Deepgram(dg_key)

options = {
    "tier": 'nova',
    "model": 'general',
    "punctuate": True,
    "diarize": True,
    "paragraphs": True,
    "detect_language": True,
    "smart_format": True
}

filename = "nLu9UU8KwqE.wav"
MIMETYPE = 'wav'

with open(filename, 'rb') as audio:
    source = {"buffer": audio, "mimetype":'audio/' + MIMETYPE}
    transcript_res = dg.transcription.sync_prerecorded(source, options)

print(transcript_res['results']['channels'][0]['alternatives'][0].keys())
```

This prints: `dict_keys(['transcript', 'confidence', 'words'])`

Usually, there is a "paragraphs" key since I included `"paragraphs": True` key in my options.

Here is the audio file I'm using: [https://drive.google.com/file/d/1XsDDiZjqxeQ--ndLjI0W6-dy6F2pUeIx/view?usp=sharing](https://drive.google.com/file/d/1XsDDiZjqxeQ--ndLjI0W6-dy6F2pUeIx/view?usp=sharing) 

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

964f0ea1-bf91-41fb-a3a2-683649d5917a

**Comment by bmaluijb** [comment]
I have the same issue. Just started happening for me with a .wav file.

Request ID: 8ac1b4cd-582c-4a9d-8e58-a7d1bf9fde06

I hope this will not be a regular occurrence.

**Comment by jkroll-deepgram** [comment]
Hi @liechtym, thanks for providing such thorough info. I ran your code against your audio file and was able to replicate this. I believe that overall the issue is submitting a larger audio file (92 MB, stereo) with a number of more intensive features (smart format, diarize, detect language). 

I noticed that your file is stereo with the same audio in each of two channels. When I ran your exact code against only one channel, then it did return paragraphs. For instance, I used the command `ffmpeg -i nLu9UU8KwqE.wav -af "pan=mono|FC=FR" mono_right.wav` to isolate the right channel. For the left channel, change `FR` to `FL`.

The team is looking into this further to determine the precise root cause and best path forward.

**Reply by maketesteasier** [reply]
Please do it fast. And please provide a temporary solution first. I also want to know the timeline of solving the issue. Otherwise, I really have to go to other apis. My product cannot function properly if I do not get paragraph.

**Reply by bmaluijb** [reply]
I'm experiencing the same issue sometimes, with short and long audio, regardless of channels. Maybe this helps for you: I'm currently using utterances as a workaround in my product. Not great, but it's something.

**Reply by maketesteasier** [reply]
can we get the timestamps of each sentence with utterances?

**Reply by bmaluijb** [reply]
Utterances have words, not sentences perse. And the utterances have a start and end time. The thing is that utterances don't match up with sentences, so typically they are smaller. For my purposes it kind of works, but I really need the paragraphs.

**Comment by liechtym** [comment]
Thanks @jkroll-deepgram.  As added information, my system has been processing audio/transcription the same way for weeks and this bug is new today. So it may be related to a new update in your code.

I wasn't aware that both channels were the same, but that is great information. Thank you.

**Comment by B2Gdevs** [comment]
Same thing happened to me.  Only happens in long files though.  Shorter files remain the same.

**Comment by maketesteasier** [comment]
I also experienced the same issue, not to all files, but to some of them. Please identify the issue as soon as possible. My product depends on your API working properly. Thanks a lot. My request id is: 1daba5b4-443c-4736-b19b-67d9aa6ffd71

**Comment by pkpio** [comment]
I'm facing the same issue. Only started last night. 1 in 4 requests keep getting this error. We are being billed for transcripts without the paragraphs info.

**Comment by jkroll-deepgram** [comment]
We are still actively investigating this issue. It seems to be principally affecting Nova, so as a temporary workaround, consider switching to Enhanced or Base. Longer/larger files are also more prone to being affected, so it may help to chunk long audio files into several shorter ones, or send only one channel from stereo audio with two identical channels.

**Reply by pkpio** [reply]
I'm using whisper-large and still seeing the issue. Keeps happening over and over despite many retries. Sadly, we are being charged for each retry!

**Comment by jkroll-deepgram** [comment, marked as answer]
We have identified the root cause and will be deploying a fix. Expected to be resolved by later today.

**Reply by pkpio** [reply]
Is this for all models?

**Reply by jkroll-deepgram** [reply]
Yes, there is one model-agnostic root cause, and the fix will apply to all models.

**Reply by maketesteasier** [reply]
You are amazing. Please update here once you guys push the update.

**Reply by maketesteasier** [reply]
My issue has been solved. You are a lifesaver. Totally. Thank you for the amazing work. Deepgram is the best speech to text tool available. I compared a lot of them and finally settled down on deepgram. Thanks again.

**Reply by maketesteasier** [reply]
Update: the files yesterday were returned with the paragraph infor today. However, I experienced this issue with a new mp3 file. It seems that the issue still exists.

**Reply by jkroll-deepgram** [reply]
@maketesteasier There was a period of a few hours from ~21:00-23:15 UTC yesterday when we were doing a rolling deployment of the fix, so you may have hit some servers that had been fixed, and others that hadn't. Please do report any new occurrences of this issue going forward, but I don't expect it to occur again.

**Comment by liechtym** [comment]
Thank you @jkroll-deepgram. I confirmed this resolved our issue. Thanks for your help.

**Comment by jkroll-deepgram** [comment]
The deployment of the fix was finished as of 23:15 UTC yesterday. Thanks all for your patience as we resolved this issue.

**Reply by bmaluijb** [reply]
Well done team!

**Comment by onyii5119** [comment]
It seems the issue has not been resolved. Paragraphs is still not showing in my result.
I am using node.js
"@deepgram/sdk": "^2.4.0",
paragraphs set to true.

**Reply by pkpio** [reply]
I noticed this happens every time I use Chinese as the output language. With Whisper large. Probably happening in other cases too.

**Reply by jkroll-deepgram** [reply]
Hi @onyii5119 and @pkpio, can you please provide any request IDs for recent requests that have been missing paragraphs? There's not currently a known issue still ongoing, so a request ID will enable me to attempt to replicate what you're experiencing and to identify which conditions are causing it to occur.

**Comment by onyii5119** [comment]
Thank you @jkroll-deepgram for responding.
This is the request ID for my last run today. 
Request ID: b602130b-7981-4b31-9220-57b6b228ff94

**Reply by jkroll-deepgram** [reply]
Thanks @onyii5119. I will investigate this and follow up within the next few days.

**Reply by jkroll-deepgram** [reply]
I took a look at this request, and I wasn't able to reproduce missing paragraphs. Confirming that you are you looking for them in the response under `response["results"]["channels"][0]["alternatives"][0]["paragraphs"]`? Are you seeing the paragraphs always missing when making repeated requests with the same audio file and request parameters, or is it an intermittent issue for you?

**Reply by onyii5119** [reply]
Thank you so much, @jkroll-deepgram.
I took a look at my API return values. I noticed that I was returning "result.results.channels[0].alternatives[0].transcript" which does not contain paragraphs. By returning "result.results.channels[0].alternatives[0].paragraphs" (which based on your feedback), I was able to see paragraphs in my response.
Thanks again.

**Reply by jkroll-deepgram** [reply]
I'm glad to hear that worked for you! Happy to help :)
